### PR TITLE
Money mask

### PR DESCRIPTION
**Pull request by roxdavirox**
Features:
- added a new package: https://www.npmjs.com/package/react-number-format - (I think that this is a good component)
- implemented format state management and form submit with valid value
- renders home page currency formatted

### CODE REVIEW

File: src/features/home/Home.js [hunk 1]
```diff
@@ -91,7 +92,7 @@ export default function Home() {
         );
       });
   };
-
+  console.log(monthlyBudgetCalcs)
```
Comment (Mazuh) [inline src/features/home/Home.js:95]: ⚠️

File: src/features/monthly-budget/BudgetForm.js [hunk 2]
```diff
@@ -48,8 +50,17 @@ export default function BudgetForm({
     }
   }, [onFormInit]);
 
+  const handleSubmit = (event) => {
+    if (!amountCurrency.value) {
+      return;
+    }
+
+    event.persist();
+    onSubmit(event, amountCurrency.value);
```
Comment (Mazuh) [inline src/features/monthly-budget/BudgetForm.js:59]: keep onSubmit [consistency](https://jgefroh.medium.com/why-consistency-is-one-of-the-top-indicators-of-good-code-352ba5d62020), **it should receive all data within a single data structure** or all data spreaded in multiple arguments, but not mixed approaches like this (receiving a single data arbitrarily separated while the rest is attached on the event).

in this commit I did something similar to `handleSubmit`: https://github.com/Mazuh/frifim/pull/5/commits/eee0d8b07d6557aec4b608fd7df1ab09b0b020af

**Review by Mazuh — CHANGES_REQUESTED** [review]
pretty good behavior, we just need some small API changes.

File: src/features/monthly-budget/BudgetTable.js [hunk 3]
```diff
@@ -37,7 +38,18 @@ export default function BudgetTable({
...
-                <span className="text-muted">R$</span> {budget.amount}
+                <NumberFormat
+                  value={Number(budget.amount)}
+                  defaultValue={Number(budget.amount)}
+                  displayType="text"
+                  fixedDecimalScale
+                  decimalSeparator={','}
+                  thousandSeparator={'.'}
+                  decimalScale={2}
+                  renderText={value => (
```
Comment (rcmuniz1994) [inline src/features/monthly-budget/BudgetTable.js:49]: It's just a suggestion. What you think about creating a function a use it here? It's just for avoid the code repetition. Something like this:

```js
const renderText = value => (<><span className="text-muted">R$</span> {value}</>)
```

File: src/features/monthly-budget/BudgetTable.js [hunk 4]
```diff
@@ -85,7 +97,18 @@ export default function BudgetTable({
...
-              <span className="text-muted">R$</span> {total}
+              <NumberFormat
+                  value={Number(total)}
+                  defaultValue={Number(total)}
+                  displayType="text"
+                  fixedDecimalScale
+                  decimalSeparator={','}
+                  thousandSeparator={'.'}
+                  decimalScale={2}
+                  renderText={value => (
```
Comment (rcmuniz1994) [inline src/features/monthly-budget/BudgetTable.js:108]: And you could use the function that I mentioned before here too. :D

File: src/features/home/Home.js [hunk 5]
```diff
@@ -122,16 +123,38 @@ export default function Home() {
...
-                    </span>
+                    <NumberFormat
+                      value={Number(monthlyBudgetCalcs.total)}
+                      defaultValue={Number(monthlyBudgetCalcs.total)}
+                      displayType="text"
+                      fixedDecimalScale
+                      decimalSeparator={','}
+                      thousandSeparator={'.'}
+                      decimalScale={2}
+                      renderText={value => (
```
Comment (rcmuniz1994) [inline src/features/home/Home.js:134]: I think you could do here the same thing I mentioned on the following comments. What do you think?

File: src/features/home/Home.js [hunk 6]
```diff
@@ -122,16 +123,38 @@ export default function Home() {
...
-                    </span>
+                    <NumberFormat
+                      value={Number(transactionsCalcs.total)}
+                      defaultValue={Number(transactionsCalcs.total)}
+                      displayType="text"
+                      fixedDecimalScale
+                      decimalSeparator={','}
+                      thousandSeparator={'.'}
+                      decimalScale={2}
+                      renderText={value => (
```
Comment (rcmuniz1994) [inline src/features/home/Home.js:152]: I think you could do here the same thing I mentioned on the following comments. What do you think?

File: src/features/monthly-budget/BudgetTable.js (hunk 3 above)
Comment (roxdavirox) [reply src/features/monthly-budget/BudgetTable.js:49]: Okay, I"ll encapsulate this. but, the other comments need a different data type. I will create a HOC that receives the dependency data

**Review by Mazuh — APPROVED** [review]

**Pull request merged**